semgrep-core/src/engine/Match_extract_mode.ml

```diff
@@ -141,6 +141,15 @@ let mk_extract_target dst_lang contents rule_ids =
     rule_nums = List.mapi (fun i _ -> i) rule_ids;
   }
 
+(* Unescapes JSON array string *)
+let extract_code_from_json json =
+  let json' = "{ \"semgrep_fake_payload\": " ^ json ^ "}" in
+  Yojson.Basic.from_string json'
+  |> Yojson.Basic.Util.member "semgrep_fake_payload"
+  |> Yojson.Basic.Util.to_list
+  |> Common.map (fun x -> x |> Yojson.Basic.Util.to_string)
+  |> String.concat "\n"
```

Comment:
I also think the name of this function could be improved to be more descriptive of what it actually is doing, given it's not really doing anything to get the code from the json, just processing the array.

Reply:
Updated this as suggested. I kept the "semgrep_fake_payload" trick because it's necessary. If not the extracted content is not a valid JSON and it can't be processed by Yojson.